pkg/jobcreator/run.go

```diff
@@ -46,20 +49,42 @@ func RunJob(
 	// wait a short period because we've just started the job creator service
 	time.Sleep(100 * time.Millisecond)
 
+	// Start run job trace
+	c, span := jobCreatorService.controller.tracer.Start(ctx.Ctx, "run_job",
+		trace.WithAttributes(
+			attribute.String("job_offer.job_creator", offer.JobCreator),
+			attribute.String("job_offer.module.repo", offer.Module.Repo),
+			attribute.String("job_offer.module.hash", offer.Module.Hash),
+			attribute.String("job_offer.mode", string(offer.Mode)),
+		))
+	ctx.Ctx = c
+	defer span.End()
+
+	span.AddEvent("add_job_offer.start")
 	jobOfferContainer, err := jobCreatorService.AddJobOffer(offer)
 	if err != nil {
 		jobCreatorService.controller.log.Error("failed to add job offer", err)
+		span.SetStatus(codes.Error, "failed to add job offer")
+		span.RecordError(err)
 		return nil, err
 	}
 	jobCreatorService.controller.log.Debug("job offer ID", jobOfferContainer.ID)
+	span.AddEvent("add_job_offer.done",
+		trace.WithAttributes(
+			attribute.String("job_offer_container.deal_id", jobOfferContainer.DealID),
+			attribute.String("job_offer_container.state", data.GetAgreementStateString(jobOfferContainer.State)),
+		))
+	span.SetAttributes(attribute.String("job_offer.id", jobOfferContainer.JobOffer.ID),
+		attribute.String("deal.id", jobOfferContainer.DealID))
```

Comment:
Setting span attributes this far into a trace is discouraged, but we need to submit the job offer before we can establish a deal ID. We want the deal ID to query on and associate with traces from other services.